lib/main.coffee

```diff
@@ -0,0 +1,53 @@
+{CompositeDisposable} = require 'atom'
+
+module.exports =
+  config:
+    shellcheckExecutablePath:
+      type: 'string'
+      title: 'Shellcheck Executable Path'
+      default: 'shellcheck' # Let OS's $PATH handle the rest
+    enableNotice:
+      type: 'boolean'
+      title: 'Enable Notice Messages'
+      default: false
+
+  activate: ->
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe \
+     'linter-shellcheck.shellcheckExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-shellcheck.enableNotice',
+      (enableNotice) =>
+        @enableNotice = enableNotice
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    helpers = require('atom-linter')
+    provider =
+      grammarScopes: ['source.shell']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        filePath = textEditor.getPath()
+        command = @executablePath
+        showAll = @enableNotice
+        parameters = []
+        parameters.push('-f', 'gcc', filePath)
+        return helpers.exec(command, parameters).then (output) ->
```

Comment:
How about you use `@executablePath` directly instead of assigning it to a variable and then using it.